lib/ts/recipe/passwordless/utils.ts

```diff
@@ -41,19 +44,73 @@ export function validateAndNormaliseUserInput(
         ...config.override,
     };
 
+    let emailService =
+        config === undefined || config.emailDelivery === undefined
+            ? undefined
+            : undefined || config.emailDelivery.service;
+    /**
+     * following code is for backward compatibility.
+     * if user has not passed emailDelivery config, we
+     * use the createAndSendCustomEmail config. If the user
+     * has not passed even that config, we use the default
+     * createAndSendCustomEmail implementation
+     */
+    if (emailService === undefined) {
+        emailService = {
+            sendEmail: async (input: TypeEmailDeliveryTypeInput & { userContext: any }) => {
+                if (config.contactMethod === "EMAIL" || config.contactMethod === "EMAIL_OR_PHONE") {
```

Comment:
This IF check is not required, since this function is only called if the content method is EMAIL or EMAIL_OR_PHONE

Reply:
In case this IF check is needed from a types point of view, then keep it.

Reply:
required from types point of view

Reply:
ok. resolving